app/services/variables_fetch_service.rb

```diff
@@ -1,16 +1,36 @@
 # frozen_string_literal: true
 
 class VariablesFetchService
+  # Load all variables from the OpenFisca server into the database
+  #
+  # Updates existing database records but does not remove old records no longer
+  # mentioned by the OpenFisca server
+  #
+  # Returns the loaded Variables, as a streamed enumerable
   def self.fetch_all
-    json_response = of_conn.get('variables').body
-    json_response.keys.each do |name|
-      variable = Variable.find_or_initialize_by(name: name)
+    variables_list.each do |v|
+      # Find or create the Variable model
+      variable = Variable.find_or_initialize_by(name: v.first)
+      # Update the model with attributes retrieved from the server
+      # Note that the href value is available here but is not populated in .fetch below
+      variable.update(v.second)
+      # Fetch the additional attributes from the server and save to the database
       fetch(variable)
+      yield variable
     end
   end
 
+  # Load the full data of a variable into the database
+  #
+  # The parameter is a Variable containing the short info (only :name is
```

Comment:
Paramater is a domain specific word in this context, maybe use a synonym?

Reply:
resolved with overkill - now uses the @param and @return conventions for Yard documentation